### PR TITLE
Fixed #29835 -- Made RelatedFieldListFilter respect ModelAdmin.ordering.

### PR DESCRIPTION
**Pull request by hramezani**
Fix for [ticket 29835](https://code.djangoproject.com/ticket/29835).

### CODE REVIEW

File: django/db/models/fields/reverse_related.py [hunk 1]
```diff
@@ -123,7 +123,7 @@ def get_choices(self, include_blank=True, blank_choice=BLANK_CHOICE_DASH):
         initially for utilization by RelatedFieldListFilter.
         """
         return (blank_choice if include_blank else []) + [
-            (x.pk, str(x)) for x in self.related_model._default_manager.all()
+            (x.pk, str(x)) for x in self.related_model._default_manager.all().order_by(*ordering)
```
Comment (charettes) [inline django/db/models/fields/reverse_related.py]: You can drop the `.all()` here.

Comment (hramezani) [reply django/db/models/fields/reverse_related.py]: Thanks

**Comment by timgraham** [comment]
Please add some test of the `get_choices()` changes in `tests/model_fields`. Without that, there would only be testing of that API through the admin tests.

**Comment by hramezani** [comment]
@timgraham thanks for the comment. required tests added.

**Pull request merged**